### PR TITLE
Search request added. API key removed 

### PR DESCRIPTION
**Pull request by BlessP**
API Set up:
- Get new API key off of Cloud Console Credentials section
- use command "export DEVELOPER_KEY=12345;mvn package appengine:run" (or deploy)

This ensures the environment variable will contain the correct key every time.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/YoutubeApi.java [hunk 1]
```diff
@@ -55,32 +55,49 @@ private static YouTube getService() throws GeneralSecurityException, IOException
 
   public static List<YoutubePost> getYoutubePost() throws YoutubeApiException {
     List<YoutubePost> list = new ArrayList();
-    List<CommentData> commentList = new ArrayList();
-    String id = "Ks-_Mh1QhMc";
+    String id;
+    String key = (System.getenv("DEVELOPER_KEY"));
```
Comment (ginellegaisano) [inline src/main/java/com/google/sps/servlets/YoutubeApi.java]: nit: remove surrounding brackets.

File: src/main/java/com/google/sps/servlets/YoutubeApi.java [hunk 2]
```diff
@@ -55,32 +55,49 @@ private static YouTube getService() throws GeneralSecurityException, IOException
...
-                                                .setTextFormat("plainText");
-      CommentThreadListResponse commentsResponse =
-          request.setKey(DEVELOPER_KEY).setMaxResults(15L).execute();
-      for (CommentThread commentThread : commentsResponse.getItems()) {
-        CommentSnippet snippet = commentThread.getSnippet().getTopLevelComment().getSnippet();
-        commentList.add(new CommentData(snippet.getTextDisplay(), snippet.getLikeCount()));
+      // Returns top posts relevant to GoogleFi and feeds those Ids into video response
+      YouTube.Search.List searchRequest = getService()
+                                              .search()
+                                              .list("snippet")
```
Comment (ginellegaisano) [inline src/main/java/com/google/sps/servlets/YoutubeApi.java]: Turn "snippet" into a static string constant and use throughout

File: src/main/java/com/google/sps/servlets/YoutubeApi.java [hunk 3]
```diff
@@ -55,32 +55,49 @@ private static YouTube getService() throws GeneralSecurityException, IOException
...
+                                              .setType("video")
+                                              .setMaxResults(3L)
```
Comment (ginellegaisano) [inline src/main/java/com/google/sps/servlets/YoutubeApi.java]: Also turn this into a constant (MAX_VIDEO_RESULTS)

File: src/main/java/com/google/sps/servlets/YoutubeApi.java [hunk 4]
```diff
@@ -55,32 +55,49 @@ private static YouTube getService() throws GeneralSecurityException, IOException
...
+        }
+        Video video = response.getItems().get(0);
+        // Retrieving comments before creating new post
+        YouTube.CommentThreads.List request = getService()
+                                                  .commentThreads()
+                                                  .list("snippet")
+                                                  .setVideoId(video.getId())
+                                                  .setTextFormat("plainText");
+        CommentThreadListResponse commentsResponse =
+            request.setKey(key).setOrder("relevance").setMaxResults(3L).execute();
```
Comment (ginellegaisano) [inline src/main/java/com/google/sps/servlets/YoutubeApi.java]: Same here - MAX_COMMENT_RESULTS

File: src/main/java/com/google/sps/servlets/YoutubeApi.java [hunk 5]
```diff
@@ -55,32 +55,49 @@ private static YouTube getService() throws GeneralSecurityException, IOException
...
+        List<CommentData> commentList = new ArrayList();
+        for (CommentThread commentThread : commentsResponse.getItems()) {
+          CommentSnippet snippet = commentThread.getSnippet().getTopLevelComment().getSnippet();
+          commentList.add(new CommentData(snippet.getTextDisplay(), snippet.getLikeCount()));
+        }
+        /*Post model (Title, Description, Video id, Video comments, video comment likes, Video
```
Comment (ginellegaisano) [inline src/main/java/com/google/sps/servlets/YoutubeApi.java]: remove comment.

**Review by jonaner — APPROVED** [review]

**Review by ginellegaisano — APPROVED** [review]

**Pull request merged**